lightsaber/gradle-plugin/src/main/java/com/joom/lightsaber/plugin/AndroidLightsaberPlugin.kt

```diff
@@ -28,14 +38,106 @@ class AndroidLightsaberPlugin : BaseLightsaberPlugin() {
       throw GradleException("Lightsaber plugin must be applied *AFTER* Android plugin")
     }
 
+    addDependencies(JavaPlugin.IMPLEMENTATION_CONFIGURATION_NAME)
+
     val extension = project.extensions.create("lightsaber", AndroidLightsaberPluginExtension::class.java)
-    val transform = LightsaberTransform(extension)
+    val componentsExtension = project.androidComponents
 
-    addDependencies(JavaPlugin.IMPLEMENTATION_CONFIGURATION_NAME)
+    if (componentsExtension != null && componentsExtension.pluginVersion >= VARIANT_API_REQUIRED_VERSION) {
+      logger.info("Registering lightsaber with variant API")
+
+      configureTransformWithComponents()
+    } else {
+      logger.info("Registering lightsaber with transform API")
```

Comment:
I think we  should log this message as `warn` with some details about deprecation of the transform

Reply:
What do you think the message should warn about? Ask user to upgrade AGP because Transform API is deprecated?